plugins/needs_restarting.py

```diff
@@ -40,6 +40,113 @@
 NEED_REBOOT = ['kernel', 'glibc', 'linux-firmware', 'systemd', 'dbus',
                'dbus-broker', 'dbus-daemon']
 
+class DirectoryOption(object):
+    """ This class handles a directory containing file options
+        This class can sanitize dir path
+    """
+    def __init__(self, path=None):
+        self._path = self._parse(path)
+
+    def _parse(self, path):
+        if not isinstance(path, str):
+            return set(path)
+        strval = path.replace('\n', ' ')
+        strval = path.replace(',', ' ')
+        strval = { x for x in strval.split(' ') if x }
+        strval = { x for x in strval if self._is_dir(x) }
```

Comment:
Overwriting this generically-named variable is not very nice. It's better to not store different things into the same variable. It can lead to errors such as overwriting the variable without actually using it's value in the first place :sunglasses: like in this case, on line 54 you don't call `strval.replace()`, so you lose the replacement of `\n` you've done on line 53.

How about this here:
```python
replaced = path.replace('\n', ' ').replace(',', ' ')
return { x for x in replaced.split() if self._is_dir(x) }
```
Note that `split()` without an argument splits on any whitespace and won't include empty strings in the result.

Again, see the whole review though.